tests/model_test.js

```diff
@@ -0,0 +1,506 @@
+        await user.destroy();
+        expect(await User.findOne({ where: 1, paranoid: false })).to.exist;
+        expect(await User.findByPk(1)).to.be.null;
+        expect(await User.count()).to.equal(2);
+        expect(await User.count({ paranoid: false })).to.equal(3);
+      });
+    });
+
+    it('should include deleted associated records if include has paranoid marked as false', async function() {
+      const User = this.sequelize.define('User', {
+        username: Sequelize.STRING
+      }, { paranoid: true });
+      const Pet = this.sequelize.define('Pet', {
+        name: Sequelize.STRING,
+        UserId: Sequelize.INTEGER
+      }, { paranoid: true });
+
+      User.hasMany(Pet);
+      Pet.belongsTo(User);
+
+      await User.sync({ force: true });
+      await Pet.sync({ force: true });
+
+      const userId = (await User.create({ username: 'Joe' })).id;
+      await Pet.bulkCreate([
+        // Added gapless incremental ids
+        { id: 1, name: 'Fido', UserId: userId },
+        { id: 2, name: 'Fifi', UserId: userId }
+      ]);
+      const pet = await Pet.findByPk(1);
+      await pet.destroy();
+      const user = await User.findOne({
+        where: { id: userId },
+        include: Pet
+      });
+      const userWithDeletedPets = await User.findOne({
+        where: { id: userId },
+        include: { model: Pet, paranoid: false }
+      });
+      expect(user).to.exist;
+      expect(user.Pets).to.have.length(1);
+      expect(userWithDeletedPets).to.exist;
+      expect(userWithDeletedPets.Pets).to.have.length(2);
+    });
+  });
+
+  describe('schematic support', () => {
+    beforeEach(async function() {
+      this.UserPublic = this.sequelize.define('UserPublic', {
+        age: Sequelize.INTEGER
+      });
+
+      this.UserSpecial = this.sequelize.define('UserSpecial', {
+        age: Sequelize.INTEGER
+      });
+
+      await Support.dropTestSchemas(this.sequelize);
+      await this.sequelize.createSchema('schema_test');
+      await this.sequelize.createSchema('special');
+      this.UserSpecialSync = await this.UserSpecial.schema('special').sync({ force: true });
+    });
+
+    afterEach(async function() {
+      try {
+        await this.sequelize.dropSchema('schema_test');
+      } finally {
+        await this.sequelize.dropSchema('special');
+        await this.sequelize.dropSchema('prefix');
+      }
+    });
+
+    // Edited test reason: Postgres expected length is 2, being ['schema_test', 'special']
+    // CRDB schemas are: ['crdb_internal', 'schema_test', 'special']
+    it('should be able to list schemas', async function() {
+      const schemas = await this.sequelize.showAllSchemas();
+      expect(schemas).to.be.instanceof(Array);
+      // originally it expected .to.have.length(expectedLength[dialect]); and being dialect postgres, length = 2.
+      expect(schemas).to.deep.equal(['crdb_internal', 'schema_test', 'special'])
+    });
+
+    // Skipped test: CRDB does not work with sequences. 
```

Comment:
This comment is misleading I think. We can remove it or indicate that by default, cockroachdb uses a UUID instead of sequence for default value.

Reply:
I think it is actually a SERIAL instead of a UUID. Either way, I'm changing the comment.